### PR TITLE
CAL-385 Address Sonarqube critical code smell findings

### PR DESCRIPTION
**Pull request by mcalcote**
#### What does this PR do?
Address the critical code smells found in Alliance by sonarqube. 

A majority of these are reducing the cognitive complexity of methods. The primary strategy behind these changes was to inspect condition logic in these methods for efficiency and to extract logic into private methods where possible.

#### Who is reviewing it? 
@troymohl @bdeining @emmberk @josephthweatt 

#### Choose 2 committers to review/merge the PR.
@coyotesqrl @rzwiefel 

#### How should this be tested?
Inspect changes, full build, test base Alliance functionality.

#### Any background context you want to provide?
These findings were reported here: https://sonarcloud.io/project/issues?id=org.codice%3Aalliance&resolved=false&severities=CRITICAL&types=CODE_SMELL

#### What are the relevant tickets?
[CAL-385](https://codice.atlassian.net/browse/CAL-385)

### CODE REVIEW

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/AccessManagerImpl.java [hunk 1]
```diff
@@ -160,11 +158,15 @@ public int get_timeout(Request aRequest)
 
   @Override
   public void set_timeout(Request aRequest, int new_lifetime)
-      throws InvalidInputParameter, ProcessingFault, SystemFault {}
+      throws InvalidInputParameter, ProcessingFault, SystemFault {
+    // This method is not expected to be called
```
Comment (bdeining) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/AccessManagerImpl.java]: @troymohl would it be more appropriate to throw `NO_IMPLEMENT` here instead of a comment?

https://github.com/mcalcote/alliance/blob/master/catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/LibraryImpl.java#L310

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfImageTransformer.java [hunk 2]
```diff
@@ -49,6 +49,9 @@
 
   private static final String IMAGE_DATATYPE = DataType.IMAGE.toString();
 
+  public static final String SETTING_THE_METACARD_ATTRIBUTE_TO =
```
[the remark this answers is not in the document]
Comment (mcalcote) [reply catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfImageTransformer.java]: No, it should be private. The default `ctrl+alt+C` makes public constants and I forgot to change 👍

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/SubmitStandingQueryRequestImpl.java [hunk 3]
```diff
@@ -640,6 +569,91 @@ protected DAGQueryResult getData(long queryTime) {
...
+      return catalogQueryRequest;
+    }
+
+    private Filter getFilter(long queryTime) {
+      Filter parsedFilter = bqsFilter;
+      if (!moreResultsAvailOnLastQuery && queryTime > 0) {
+        parsedFilter =
+            filterBuilder.allOf(
+                bqsFilter,
+                filterBuilder.attribute(Metacard.MODIFIED).is().after().date(new Date(queryTime)));
```
[the remark this answers is not in the document]
Comment (mcalcote) [reply catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/SubmitStandingQueryRequestImpl.java]: Unfortunately, `Core` doesn't provide the `DEFAULT_TAG` "resource" , so I can't replace all of them. Since I can't replace them all, I don't know if we should replace the others... Who else would want to weigh in on this? Maybe @brendan-hofmann ?

File: distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/SampleNsiliClient.java [hunk 4]
```diff
@@ -376,113 +385,111 @@ public String testOrder(DAG dag) throws Exception {
...
+      if (emailAddress != null) {
+        order = createEmailOrder(orb, product, supportedPackageSpecs);
+
+        // Validating Order
+        LOGGER.info("Validating Email Order...");
+        validationResults = orderMgr.validate_order(order, properties);
+
+        LOGGER.info("Email Validation Results: ");
+        LOGGER.info(
+            "\tValid : "
```
[the remark this answers is not in the document]
Comment (mcalcote) [reply distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/SampleNsiliClient.java]: Didn't come up as a "code smell" but I am sure IJ would recommend changing it. 👍

File: catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/NsiliCommonUtils.java [hunk 5]
```diff
@@ -87,19 +87,22 @@ public static void setUCOEdges(Node root, Graph<Node, Edge> graph) {
...
         }
       }
     }
   }
+
+  private static void processEdges(Graph<Node, Edge> graph, Stack<Node> stack, Edge edge) {
+    Node source = graph.getEdgeSource(edge);
+    Node target = graph.getEdgeTarget(edge);
+
+    // Remove if statement?
```
Comment (vinamartin) [inline catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/NsiliCommonUtils.java]: ✏️  Nitpick: Can we get rid of this comment or the if statement? 
I can already feel it's starting to get stale/confusing.

Comment (mcalcote) [reply catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/NsiliCommonUtils.java]: I will remove the comment

File: catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java [hunk 6]
```diff
@@ -188,34 +188,36 @@ private boolean hasValidLocation(String location) {
...
-            // ignore
+    if (derivedResourceUriAttribute == null) {
+      return Optional.empty();
+    }
+    List<String> derivedResourceUriStrings =
+        derivedResourceUriAttribute
+            .getValues()
+            .stream()
+            .filter(String.class::isInstance)
+            .map(Object::toString)
```
Comment (rzwiefel) [inline catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java]: ❓ Did you mean `String.class::cast`?

File: catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java [hunk 7]
```diff
@@ -188,34 +188,36 @@ private boolean hasValidLocation(String location) {
...
+        }
+        for (NameValuePair parameter :
+            URLEncodedUtils.parse(derivedResourceUri, StandardCharsets.UTF_8.name())) {
+          if (QUALIFIER_KEY.equals(parameter.getName())
+              && StringUtils.equals(ORIGINAL_QUALIFIER, parameter.getValue())) {
+            return Optional.of(derivedResourceUri);
           }
         }
+      } catch (URISyntaxException e) {
+        // ignore
```
Comment (rzwiefel) [inline catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java]: ✏️  Are we sure we don't want to log it... even at trace or something?

Comment (emmberk) [reply catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java]: This is not an unexpected exception. If we get a `URISyntaxException` here, it means that the `resource.derived-uri` attribute value does not match the usual Alliance download URL format, so we don't have enough information to construct a chipping URL. It might be worthwhile to add a log statement just before `return Optional.empty();` so that all situations that drop down to that line will log the same thing.

File: catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/datamodel/NsiliDataModel.java [hunk 8]
```diff
@@ -1342,24 +1349,25 @@ public EntityGraph getEntityGraph(String viewName) {
...
-              List<String> attrs = attrMap.get(parentNode);
-              if (attrs == null) {
-                attrs = new ArrayList<>(4);
-                attrMap.put(parentNode, attrs);
-              }
-              attrs.add(attrName);
-            }
+    List<AttributeInformation> nodeAttrs =
+        Arrays.stream(viewNodes)
+            .map(e -> getAttributeInformation(e.entity_name))
```
Comment (rzwiefel) [inline catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/datamodel/NsiliDataModel.java]: ✏️ Couldn't this also be (just a nit, no need to change if you don't care to)
```java
List<AttributeInformation> nodeAttrs = Arrays.stream(viewNodes)
  .map(e -> e.entity_name) //is there a getter so we could do .map(e::getEntity_name)?
  .map(this::getAttributeInformation)
...
```

File: catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/datamodel/NsiliDataModel.java [hunk 9]
```diff
@@ -1342,24 +1349,25 @@ public EntityGraph getEntityGraph(String viewName) {
...
+        String[] attrNameArr = attributeName.split("\\.");
+        if (attrNameArr.length == 2) {
+          String parentNode = attrNameArr[0];
+          String attrName = attrNameArr[1];
+          List<String> attrs = attrMap.get(parentNode);
+          if (attrs == null) {
+            attrs = new ArrayList<>(4);
+            attrMap.put(parentNode, attrs);
           }
+          attrs.add(attrName);
```
Comment (rzwiefel) [inline catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/datamodel/NsiliDataModel.java]: please ignore. just was doing for my own visualization/comparison

```java
List<AttributeInformation> nodeAttrs = Arrays.stream(viewNodes)
  .map(e -> e.entity_name)
  .map(this::getAttributeInformation)
  .filter(Objects::nonNull)
  .flatMap(List::stream)
  .filter(nodeAttr -> nodeAttr.mode == RequirementMode.MANDATORY)
  .map(nodeAttr -> nodeAttr.attribute_name
  .map(attrName -> attrName.split("\\."))
  .filter(attrNameArr -> attrNameArr.length == 2)
  .forEach(attrNameArr -> {
    String parentNode = attrNameArr[0];
    String attrName = attrNameArr[1];
    List<String> attrs = Optional.of(attrMap)
      .map(map -> map.get(parentNode);
      .orElse(new ArrayList<>(4);
    attrMap.put(parentNode, attrs);
    attrs.add(attrName);
  });
```

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/AccessManagerImpl.java (hunk 1 above)
Comment (troymohl) [reply catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/AccessManagerImpl.java]: I don't recall off the top of my head, but I think some of these empty methods don't throw exceptions because there was something calling those methods that couldn't handle a NO_IMPLEMENT exception.

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/SubmitStandingQueryRequestImpl.java [hunk 10]
```diff
@@ -461,17 +464,17 @@ public void run() {
...
                 }
               } catch (InvalidInputParameter invalidInputParameter) {
-                LOGGER.debug("Unable to notify callback", invalidInputParameter);
+                LOGGER.debug(UNABLE_TO_NOTIFY_CALLBACK, invalidInputParameter);
               } catch (ProcessingFault processingFault) {
-                LOGGER.debug("Unable to notify callback", processingFault);
+                LOGGER.debug(UNABLE_TO_NOTIFY_CALLBACK, processingFault);
               } catch (SystemFault systemFault) {
-                LOGGER.debug("Unable to notify callback", systemFault);
+                LOGGER.debug(UNABLE_TO_NOTIFY_CALLBACK, systemFault);
```
Comment (troymohl) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/SubmitStandingQueryRequestImpl.java]: Since the handling of these are all the same, could change them to a single catch.

File: distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/Client.java [hunk 11]
```diff
@@ -118,6 +85,43 @@ public void runTests(String iorUrl, String emailAddress) throws Exception {
...
+        LOGGER.info("Unable to test ProductMgr: {}", NsilCorbaExceptionUtil.getExceptionDetails(e));
+      }
+      LOGGER.info("-----------------------");
+
+      // OrderMgr
+      PackageElement[] packageElements = sampleNsiliClient.order(results[index]);
+
+      // ProductMgr
+      if (SHOULD_PROCESS_PKG_ELEMENTS) {
+        // TODO prod is null for packageElement https://codice.atlassian.net/browse/CAL-192
```
Comment (troymohl) [inline distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/Client.java]: It looks like the issue referenced here is completed. Maybe this comment can be removed? @emmberk

File: distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/SampleNsiliClient.java [hunk 12]
```diff
@@ -146,6 +146,15 @@
   private static final Query STANDING_ALL_QUERY =
       new Query(NsiliConstants.NSIL_ALL_VIEW, "NSIL_CARD.identifier like '%'");
 
+  private static final String TAB = "\t {}";
```
Comment (troymohl) [inline distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/SampleNsiliClient.java]: Might be useful to give this a more descriptive name like `TAB_LOG_MSG`

File: distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/Client.java (hunk 11 above)
Comment (emmberk) [reply distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/Client.java]: I don't remember why I added this comment. It looks like I noted that `packageElement.prod` will always be `null`. The comment can probably just be removed.

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/OrderRequestImpl.java [hunk 13]
```diff
@@ -532,114 +378,251 @@ private boolean isEmail(Destination dest) {
...
+        }
+        break;
+      case TARCOMPRESS:
+        {
+          writeTarFile(
+              destinationSink, packagingSpecFormatType, filename, sentFiles, file, metacards);
+        }
+        break;
+      default:
+        break;
```
Comment (rzwiefel) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/OrderRequestImpl.java]: ✏️ Maybe we could do in a different PR, but should we be logging when we hit this default something along the lines of "Unknown packaging type, skipping".

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/SubmitStandingQueryRequestImpl.java (hunk 10 above)
Comment (rzwiefel) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/SubmitStandingQueryRequestImpl.java]: ✏️  These could be collapsed into (We can always do in a later PR)
```java
...
} catch (InvalidInputParamter | ProcessingFault | SystemFault failureException) {
  LOGGER.debug(UNABLE_TO_NOTIFY_CALLBACK, failureException);
}
```

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/AccessManagerImpl.java (hunk 1 above)
Comment (mcalcote) [reply catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/AccessManagerImpl.java]: Will leave with just a comment for now 👍

File: catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java (hunk 6 above)
Comment (mcalcote) [reply catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java]: Yes 👍

File: catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java (hunk 7 above)
Comment (mcalcote) [reply catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java]: Will update to `This is not an unexpected exception, there is not enough info to construct a chipping URL`

File: catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/datamodel/NsiliDataModel.java (hunk 8 above)
Comment (mcalcote) [reply catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/datamodel/NsiliDataModel.java]: will update, I think it looks cleaner to have a single action per line

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/SubmitStandingQueryRequestImpl.java (hunk 10 above)
Comment (mcalcote) [reply catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/SubmitStandingQueryRequestImpl.java]: Good call, will update

File: distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/Client.java (hunk 11 above)
Comment (mcalcote) [reply distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/Client.java]: Removed

File: distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/SampleNsiliClient.java (hunk 12 above)
Comment (mcalcote) [reply distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/SampleNsiliClient.java]: Updated 👍

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/OrderRequestImpl.java (hunk 13 above)
Comment (mcalcote) [reply catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/OrderRequestImpl.java]: I can add

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/SubmitStandingQueryRequestImpl.java (hunk 10 above)
Comment (mcalcote) [reply catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/SubmitStandingQueryRequestImpl.java]: Collapsed 👍

**Comment by mcalcote** [comment]
build now
